frontend/src/metabase/dashboard/components/grid/GridLayout.jsx

```diff
@@ -95,8 +99,20 @@ export function GridLayout({
   }, [cellSize.height, layout, margin, isEditing]);
 
   const background = useMemo(
-    () => generateGridBackground({ cellSize, margin, cols, gridWidth }),
-    [cellSize, gridWidth, margin, cols],
+    () =>
+      generateGridBackground({
+        cellSize,
+        margin,
+        cols,
+        gridWidth,
+
+        // We cannot use CSS variables here, as the svg data in background-image
+        // lives a separate style tree from the rest of the app.
+        cellStrokeColor:
+          theme.other?.dashboard?.gridBorderColor ??
+          theme.fn.themeColor("border"),
```

Comment:
I tried to use CSS variables in `generateGridBackground` e.g. `style="stroke: var(--mb-color-border)"` already, but this does not work in this case because we're generating inline SVG data URIs dynamically for background-image for the grids.

https://github.com/metabase/metabase/blob/2aa3cec610588238132afbc789da0845c7c2ec5d/frontend/src/metabase/dashboard/components/grid/utils.js#L39-L50